ui/js/components/auth.js

```diff
@@ -0,0 +1,173 @@
+"use strict";
+
+/**
+ * This component handles logging in to Taskcluster Authentication
+ *
+ * See: https://docs.taskcluster.net/manual/3rdparty
+ *
+ * This communicates to the rest of Treeherder by calling the onUserChange
+ * function on the HTML element, which in turn stores that user in $rootScope.
+ */
+treeherder.component("login", {
+    template: `
+        <span class="dropdown"
+              ng-if="$ctrl.user.loggedin">
+          <button id="logoutLabel" title="Logged in as: {{$ctrl.user.email}}" role="button"
+                  data-toggle="dropdown" data-target="#"
+                  class="btn btn-view-nav btn-right-navbar">
+            <div class="nav-user-icon">
+              <span class="fa fa-user pull-left"></span>
+            </div>
+            <span class="fa fa-angle-down lightgray"></span>
+          </button>
+          <ul class="dropdown-menu" role="menu" aria-labelledby="logoutLabel">
+              <li>
+                <a ng-click="$ctrl.logout()">Logout</a>
+              </li>
+          </ul>
+        </span>
+
+        <a class="btn btn-view-nav btn-right-navbar nav-login-btn"
+           ng-if="!$ctrl.user.loggedin && $ctrl.userCanLogin"
+           ng-click="$ctrl.login()">Login/Register</a>
+        <span ng-if="!$ctrl.userCanLogin"
+              class="midgray"
+              title="thServiceDomain does not match host domain">Login not available</span>
+    `,
+    bindings: {
+        // calls to the HTML which sets the user value in the $rootScope.
+        onUserChange: "&"
+    },
+    controller: ['$location', '$window', 'localStorageService',
+        'ThUserModel', '$http', 'thUrl', '$timeout', 'thServiceDomain',
+        function ($location, $window, localStorageService,
+                  ThUserModel, $http, thUrl, $timeout, thServiceDomain) {
+            var ctrl = this;
+            ctrl.user = {};
+            // "clears out" the user when it is detected to be logged out.
+            var loggedOutUser = {is_staff: false, username: "", email: "", loggedin: false};
+
+            // check if the user can login.  thServiceDomain must match
+            // host domain.  Remove this is we fix
+            // Bug 1317752 - Enable logging in with Taskcluster Auth cross-domain
+            var a = document.createElement('a');
+            a.href = thServiceDomain;
+            this.userCanLogin = a.hostname === $location.host();
```

Comment:
Could this just check to see if `thServiceDomain` is the empty string? (iirc it's empty when `local.conf.js` isn't being used). (This would save having to create an element)